### ISSUE TITLE
Views don't properly filter on multiple keys

### ISSUE THREAD

**Issue by timritzer** [couchbase/sg-bucket#16]
Views allow several filter mechanisms, including key and keys.

Walrus views do not properly support the view keys parameter, and return all data if provided with a set of keys.

The single key filtering mechanism works as expected, as does start/end key lookups for ranges.

Tested all of the above with a full CB bucket, and the filtering works as expected.

**Comment by adamcfraser** [comment]
Is this an issue when using Sync Gateway with Walrus, Sync Gateway with Couchbase Server, or all of the above?

**Comment by timritzer** [comment]
Sync gateway with walrus. With a full CB bucket it works fine. Looking at the implementation it looks like CB does it's filtering itself. But I'm not very good at reading Erlang yet.

**Comment by adamcfraser** [comment]
That's correct - walrus doesn't currently support the `keys` parameter.  Walrus isn't intended to provide parity of view functionality with Couchbase Server - primarily just the core functionality required for the most common use cases.

What's your use case for walrus where you're running into this issue?

**Comment by timritzer** [comment]
Writing development views for eventual deployment to couchbase server. 

I would be happy with a naive implementation of simply applying the key filter repeatedly, and unioning the datasets, the speed isn't important. It just feels really wrong to just ignore the filter and return all documents like it does now. It causes Walrus to legitimately not be usable since my application logic can't work when the wrong results are returned.
